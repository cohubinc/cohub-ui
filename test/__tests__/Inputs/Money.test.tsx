import React from "react";
import { mount } from "enzyme";

import { Inputs } from "dist";

describe("Money", () => {
  it("renders Money without crashing", () => {
    const wrapper = mount(
      <Inputs.Money input={{ value: 0, onChange: _value => undefined }} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
