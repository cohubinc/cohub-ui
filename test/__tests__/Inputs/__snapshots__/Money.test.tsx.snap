// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Money renders Money without crashing 1`] = `
<MoneyInput
  input={
    Object {
      "onChange": [Function],
      "value": 0,
    }
  }
>
  <div
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <FloatingLabelWrapper
      error={false}
      onChange={[Function]}
      value={0}
    >
      <div
        className="FloatingLabelWrapper GenericInput "
        style={
          Object {
            "cursor": "text",
            "position": "relative",
          }
        }
      >
        <div
          className="inputWrapper   bd-radius"
        >
          <NumberFormat
            allowEmptyFormatting={false}
            allowLeadingZeros={false}
            allowNegative={true}
            aria-invalid={false}
            decimalScale={2}
            decimalSeparator="."
            displayType="input"
            fixedDecimalScale={false}
            getInputRef={[Function]}
            isAllowed={[Function]}
            isNumericString={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseUp={[Function]}
            onValueChange={[Function]}
            prefix="$"
            style={
              Object {
                "textAlign": "left",
              }
            }
            suffix=""
            thousandSeparator={true}
            thousandsGroupStyle="thousand"
            type="text"
            value=""
          >
            <input
              aria-invalid={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseUp={[Function]}
              style={
                Object {
                  "textAlign": "left",
                }
              }
              type="text"
              value=""
            />
          </NumberFormat>
          <span
            className="bar "
          />
        </div>
      </div>
    </FloatingLabelWrapper>
  </div>
</MoneyInput>
`;
